shanoir-ng-datasets/src/main/java/org/shanoir/ng/configuration/amqp/RabbitMQDatasetsService.java

```diff
@@ -370,4 +370,17 @@ public void deleteStudy(String eventAsString) throws AmqpRejectAndDontRequeueExc
 			throw new AmqpRejectAndDontRequeueException(RABBIT_MQ_ERROR + e.getMessage(), e);
 		}
 	}
+
+
+	@Transactional
+	@RabbitListener(queues = RabbitMQConfiguration.STUDY_DATASET_TYPE)
+	@RabbitHandler
+	public String getDatasetAcquisitionType(Long studyId) {
+		SecurityContextUtil.initAuthenticationContext("ADMIN_ROLE");
+		DatasetAcquisition datasetAcquisition = this.datasetAcquisitionService.findById(studyId);

```

Comment:
Here you are loading a dataset acqusition using the study id!
1 subject -> multiple examinations -> multiple acquisitions -> Multiple datasets
So there can be multiple acquisitions for a single study.
Here you should load all acquisitions for this given study ID and return this list of acquisitions type instead of only one.

Reply:
All of this functionality (getting modalities of studies) will be done differently in another issue